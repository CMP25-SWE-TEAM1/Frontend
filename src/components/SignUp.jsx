import React from "react"
import { useState } from "react"
import { Link } from "react-router-dom"
import ReCAPTCHA from "react-google-recaptcha"
import { Modal, Box } from "@mui/material"
import lightLogo from "../assets/imgs/giga-chat-logo-dark-removebg-preview.png"


import { styles } from "../styles"
import GoogleLoginButton from "./GoogleLoginButton"

const SignUp = ({ openModal, handleCloseModal, location, setLocation }) => {
  const [nickName, setNickName] = useState("")
  const [email, setEmail] = useState("")
  const [date, setDate] = useState("")
  const [password, setPassword] = useState("")
  const [showPassword, setShowPassword] = useState(false)

  const siteKey = "6Ldxlf4oAAAAAKjm3gXBNjq-GBJ4hM79g6NYk7KG"
  const togglePasswordVisibility = () => {
    setShowPassword(!showPassword)
  }
  function responseGoogle(response) {
    console.log(response)
  }
  const handleCaptchaVerification = (response) => {
    console.log("Captcha response: ", response)
  }
  function nextShow(position) {
    const JoinGigaChat = document.getElementById("Join GigaChat")
    const FirstStep = document.getElementById("First Step")
    const SecondStep = document.getElementById("Second Step")
    const ThirdStep = document.getElementById("Third Step")
    switch (position) {
      case 0:
        JoinGigaChat.style.display = "none"
        FirstStep.style.display = "block"
        break
      case 1:
        FirstStep.style.display = "none"
        SecondStep.style.display = "block"
        break
      case 2:
        SecondStep.style.display = "none"
        ThirdStep.style.display = "block"
        break
      default:
        break
    }
  }
  return (
    <>
      <Modal open={openModal} onClose={handleCloseModal} className="w-[90%]" disableEscapeKeyDown disablePortal>
        <Box style={styles.modalStyle}>
          <div className="pop-up m-auto bg-white dark:bg-black md:rounded-2xl">
            <Link to="/" className="!text-white" onClick={handleCloseModal}>
              <button className="bg-white dark:bg-black text-black dark:text-white hover:bg-lightHover dark:hover:bg-darkHover relative left-[-80px] top-4 h-10 w-10 rounded-3xl bg-transparent text-2xl no-underline">x</button>
            </Link>
            <img src={lightLogo} alt="GigaChat Logo" className="-mt-4 ml-[45%] w-[40px]" />

            <div id="Join GigaChat">
              <div>
                <h1 className="mb-4 mt-3">Join GigaChat today</h1>
                <GoogleLoginButton handleCloseModal={handleCloseModal} />
                <div className="flex h-10 items-center justify-center">
                  <div className="flex w-full items-center">
                    <hr className="mr-2 w-full" />
                  </div>
                  &nbsp; or &nbsp;
                  <div className="flex w-full items-center">
                    <hr className="ml-2 w-full" />
                  </div>
                </div>
                <button
                  className="btn mt-0"
                  onClick={() => {
                    nextShow(0)
                  }}
                >
                  Create Account
                </button>
                <span className="text-xs text-secondary">
                  By signing up, you agree to the <a href="/#">Terms of Service</a> and <a href="/#">Privacy Policy</a>, including <a href="/#">Cookie Use</a>.
                </span>
                <span className="mt-3 text-sm text-secondary">
                  Have an account already ?&nbsp;
                  <Link to="/login" className="text-white">
                    Log in
                  </Link>
                </span>
              </div>
            </div>

            <div id="First Step" className="hidden">
              <div>
                <p className="relative -ml-2 mt-3 text-lg font-semibold">Step 1 of 3</p>
                <h1 className="mt-3">Create your account</h1>
                <div className="input-container">
                  <input className={nickName === "" ? "form-input" : "form-input filled-input"} type="text" name="name" id="name" autoComplete="off" value={nickName} onChange={(e) => setNickName(e.target.value)} />
                  <label className="input-label" htmlFor="name">
                    Name
                  </label>
                </div>
                <div className="input-container">
                  <input className={email === "" ? "form-input" : "form-input filled-input"} type="text" name="email" id="email" autoComplete="off" value={email} onChange={(e) => setEmail(e.target.value)} />
                  <label className="input-label" htmlFor="email">
                    Email
                  </label>
                </div>
                <div className="input-containter">
                  <div>
                    <p className="text-bold">Date of birth </p>
                    <p className="date-text">This will not be shown publicly. Confirm your own age, even if this account is for a business, a pet, or something else.</p>
                    <br></br>
                  </div>
                  <input
                    type="date"
                    className="date-input text-black"
                    value={date}
                    onChange={(e) => {
                      setDate(e.target.value)
                      console.log(e.target.value)
                    }}
                  ></input>
                </div>
                <button
                  className="btn"
                  id="next"
                  onClick={() => {
                    nextShow(1)
                  }}
                  disabled={email === "" || nickName === "" || date === ""}
                >
                  Next
                </button>
              </div>
            </div>

            <div id="Second Step" className="hidden">
              <div>
                <p className="relative -ml-2 mt-3 text-lg font-semibold">Step 2 of 3</p>
                <ReCAPTCHA sitekey={siteKey} onChange={handleCaptchaVerification()} />
                <button
                  className="btn"
                  onClick={() => {
                    nextShow(2)
                  }}
                >
                  Next
                </button>
              </div>
            </div>

            <div id="Third Step" className="hidden">
              <div>
                <p className="relative -ml-2 mt-3 text-lg font-semibold">Step 3 of 3</p>
                <h1 className="">You'll need a Password</h1>
                <p className="date-text">Make sure it's 8 characters or more</p>
                <div className="relative">
                  <div className="input-container">
                    <input className={password === "" ? "form-input" : "form-input filled-input"} type={showPassword ? "text" : "password"} name="password" id="password" autoComplete="off" value={password} onChange={(e) => setPassword(e.target.value)} />
                    <label className="input-label" htmlFor="password">
                      Password
                    </label>
                  </div>
                  <span className={`toggle-password absolute right-4 top-4 cursor-pointer ${showPassword ? "active" : ""}`} onClick={togglePasswordVisibility}>
                    üëÅÔ∏è
                  </span>
                </div>
                <div className="mt-auto">
                  <Link to="/home" className="text-white">
                    <button className="btn" disabled={password.length <= 8}>
                      Next
                    </button>
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </Box>
      </Modal>
    </>
  )
}
export default SignUp
