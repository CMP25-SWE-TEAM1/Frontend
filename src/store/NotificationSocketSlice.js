import { createSlice } from "@reduxjs/toolkit"

const notificationSocketSlice = createSlice({
  name: "notificationSocket",
  initialState: {
    value: null,
  },
  reducers: {
    setNotificationSocket: (state, action) => {
      state.value = action.payload
    },
  },
})

export const { setNotificationSocket } = notificationSocketSlice.actions
export const selectNotificationSocket = (state) => state.notificationSocket.value
export default notificationSocketSlice.reducer
