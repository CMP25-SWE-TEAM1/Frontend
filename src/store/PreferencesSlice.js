import { createSlice } from "@reduxjs/toolkit"

const initMode = localStorage.getItem("preferences")

const initialState = {
  showBlockedandMuted: initMode === "true" ? true : false,
}

export const preferencesSlice = createSlice({
  name: "preferences",
  initialState,
  reducers: {
    toggleBlockedMutedMode: (state) => {
      state.showBlockedandMuted = state.showBlockedandMuted ? false : true
      localStorage.setItem("preferences", state.showBlockedandMuted ? "true" : "false")
    },
  },
})

export const toggleBlockedMutedMode = preferencesSlice.actions.toggleBlockedMutedMode

export default preferencesSlice.reducer
