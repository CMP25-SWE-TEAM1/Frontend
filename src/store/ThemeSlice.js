import { createSlice } from "@reduxjs/toolkit"

const initMode = localStorage.getItem("mode")
const initColor = localStorage.getItem("color")
console.log(initColor)

const initialState = {
  darkMode: initMode === "true" ? true : false,
  color: initColor === undefined ? 1 : parseInt(initColor),
}

export const themeSlice = createSlice({
  name: "theme",
  initialState,
  reducers: {
    setDarkMode: (state) => {
      const htmlElement = document.getElementById("htmlid")
      state.darkMode = true
      document.documentElement.style.setProperty("--color-theme", "dark")
      htmlElement.classList.add("dark")
    },
    setLightMode: (state) => {
      const htmlElement = document.getElementById("htmlid")
      state.darkMode = false
      document.documentElement.style.setProperty("--color-theme", "white")
      htmlElement.classList.remove("dark")
    },
    changeColor: (state, actions) => {
      state.color = actions.payload.color
      localStorage.setItem("color", actions.payload.color)
    },
  },
})

export const toggleTheme = themeSlice.actions
export const changeColor = themeSlice.actions.changeColor
export const setDarkMode = themeSlice.actions.setDarkMode
export const setLightMode = themeSlice.actions.setLightMode

export default themeSlice.reducer
