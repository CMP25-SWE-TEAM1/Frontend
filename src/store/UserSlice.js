import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
import axios from "axios"

// TODO store token instead of user

const APIs = {
  mock: { loginAPI: "https://ca224727-23e8-4fb6-b73e-dc8eac260c2d.mock.pstmn.io/login" },
  actual: {
    loginAPI: "http://backend.gigachat.cloudns.org/api/user/login",
  },
}

let google = false

export const loginUser = createAsyncThunk("user/loginUser", async ({ userCredentials, isgoogle }) => {
  let response
  if (isgoogle) {
    google = true
    console.log(userCredentials)
    response = userCredentials
    localStorage.setItem("user", JSON.stringify(userCredentials))
  } else {
    google = false
    const request = await axios.post(APIs.actual.loginAPI, userCredentials)
    response = await request.data
    localStorage.setItem("user", JSON.stringify(response.data.user))
  }

  return response
})

const initialUser = localStorage.getItem("user")
const initialToken = localStorage.getItem("token")

const userSlice = createSlice({
  name: "user",
  initialState: {
    loading: false,
    user: initialUser ? JSON.parse(initialUser) : null,
    error: null,
    token: initialToken ? JSON.parse(initialToken) : null,
  },
  reducers: {
    logoutUser: (state) => {
      state.user = null
      state.loading = false
      state.error = null
    },
    signupUser: (state, action) => {
      console.log(action.payload)
      state.user = action.payload.user

      //next line will be removed
      localStorage.setItem("user", JSON.stringify(action.payload.user))
      localStorage.setItem("token", JSON.stringify(action.payload.token))

      state.loading = false
      state.error = null
      state.token = action.payload.token

      action.payload.navigate("/home")
    },
    changeProfilePicture: (state, action) => {
      state.user = action.payload.user

      localStorage.setItem("user", JSON.stringify(action.payload.user))
      localStorage.setItem("token", JSON.stringify(action.payload.token))

      state.loading = false
      state.error = null
      state.token = action.payload.token
    },
    changeUsername: (state, action) => {
      state.user.username = action.payload
      let temp_user = JSON.parse(localStorage.getItem("user"))
      temp_user.username = action.payload
      localStorage.setItem("user", JSON.stringify(temp_user))
    },
    changeEmail: (state, action) => {
      state.user.email = action.payload
      let temp_user = JSON.parse(localStorage.getItem("user"))
      temp_user.email = action.payload
      localStorage.setItem("user", JSON.stringify(temp_user))
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(loginUser.pending, (state) => {
        state.loading = true
        state.user = null
        state.error = null
        state.token = null
      })
      .addCase(loginUser.fulfilled, (state, action) => {
        state.loading = false
        state.user = google ? action.payload : action.payload.data.user
        state.error = null
        state.token = action.payload.token
      })
      .addCase(loginUser.rejected, (state, action) => {
        state.loading = false
        state.user = null
        console.log(action.error.message)
        if (action.error.message === "Request failed with status code 401") {
          state.error = "Access Denied! Invalid Credentials"
        } else {
          state.error = "Access Denied! Invalid Credentials"

          //this may be changed
          // state.error = action.error.message
        }
        state.token = null
      })
  },
})

export const logoutUser = userSlice.actions.logoutUser
export const signupUser = userSlice.actions.signupUser
export const changeProfilePicture = userSlice.actions.changeProfilePicture
export const changeUsername = userSlice.actions.changeUsername
export const changeEmail = userSlice.actions.changeEmail

export default userSlice.reducer
