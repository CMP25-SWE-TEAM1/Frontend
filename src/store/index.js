import { configureStore } from "@reduxjs/toolkit"
import userReducer from "./UserSlice"
import themeReducer from "./ThemeSlice"
import socketReducer from "./SocketSlice"
import notificationSocketReducer from "./NotificationSocketSlice"
import PreferencesSlice from "./PreferencesSlice"

const store = configureStore({
  reducer: {
    user: userReducer,
    theme: themeReducer,
    socket: socketReducer,
    notification: notificationSocketReducer,
    preferences: PreferencesSlice,
  },
})

export default store
