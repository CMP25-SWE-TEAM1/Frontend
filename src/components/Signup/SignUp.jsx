import { useState } from "react"
import {  useNavigate } from "react-router-dom"

import { Modal, Box } from "@mui/material"

import { useSelector } from "react-redux"
import { useDispatch } from "react-redux"
import { loginUser, signupUser } from "../../store/UserSlice.js"

import UploadProfilePicture from "./UploadProfilePicture.jsx"
import FirstStep from "./FirstStep.jsx"
import SecondStep from "./SecondStep.jsx"
import ThirdStep from "./ThirdStep.jsx"

import { styles } from "../../styles.js"
import lightLogo from "../../assets/imgs/logo-light.jpg"
import darkLogo from "../../assets/imgs/logo-dark.jpg"

import { EMAIL_REGEX } from "../../constants/signupConstants.js"
import ForthStep from "./ForthStep.jsx"
import FifthStep from "./FifthStep.jsx"
import TagStep from "./TagStep.jsx"
import ErrorPage from "./ErrorPage.jsx"
import PreStep from "./PreStep.jsx"

const SignUp = ({ openModal, handleCloseModal }) => {
  const mock = true

  const darkMode = useSelector((state) => state.theme.darkMode)

  const dispatch = useDispatch()
  const navigate = useNavigate()

  const [nickName, setNickName] = useState("")
  const [email, setEmail] = useState("")
  const [year, setYear] = useState("")
  const [month, setMonth] = useState("")
  const [day, setDay] = useState("")
  const [userToken, setUserToken] = useState("")
  const [user, setUser] = useState()
  const [userTag, setUserTag] = useState()
  const [originalUsername, setOriginalUsername] = useState("")

  const [password, setPassword] = useState("")

  const [emailExistError, setEmailExistError] = useState(false)
  const [openBirthdateErrorModal, setOpenBirthdateErrorModal] = useState(false)

  function nextShow(position) {
    const JoinGigaChat = document.getElementById("Join GigaChat")
    const FirstStep = document.getElementById("First Step")
    const SecondStep = document.getElementById("Second Step")
    const ThirdStep = document.getElementById("Third Step")
    const ForthStep = document.getElementById("Forth Step")
    const FifthStep = document.getElementById("Fifth Step")
    const TagStep = document.getElementById("Tag Step")
    const PictureStep = document.getElementById("Picture Step")

    switch (position) {
      case 0:
        JoinGigaChat.style.display = "none"
        FirstStep.style.display = "block"
        break
      case 1:
        FirstStep.style.display = "none"
        SecondStep.style.display = "block"
        break
      case 2:
        SecondStep.style.display = "none"
        ThirdStep.style.display = "block"
        break
      case 3:
        ThirdStep.style.display = "none"
        ForthStep.style.display = "block"
        break
      case -1:
        ThirdStep.style.display = "none"
        FirstStep.style.display = "block"
        break
      case 4:
        ForthStep.style.display = "none"
        FifthStep.style.display = "block"
        break
      case 5:
        FifthStep.style.display = "none"
        TagStep.style.display = "block"
        break
      case 6:
        TagStep.style.display = "none"
        PictureStep.style.display = "block"
        break
      default:
        break
    }
  }

  function validEmail(emeil) {
    return EMAIL_REGEX.test(emeil)
  }

  const handleCloseBirthdateErrorModal = () => setOpenBirthdateErrorModal(false)
  const handleOpenBirthdateErrorModal = () => setOpenBirthdateErrorModal(true)

  const handleOpenBirthdateError = () => {
    const ErrorPage = document.getElementById("Error Page")
    const ThirdStep = document.getElementById("Third Step")

    ThirdStep.style.display = "none"
    ErrorPage.style.display = "block"

    handleOpenBirthdateErrorModal()
  }

  const handleCompleteSignup = (user) => {
    handleCloseModal()

    let userCredentials = {
      email: email,
      password: password,
    }
    // dispatch(signupUser({ user: user, token: userToken, navigate }))

    dispatch(loginUser({ userCredentials, isgoogle: null })).then((result) => {
      // console.log(result)
      if (result.payload) {
        setEmail("")
        setPassword("")
        handleCloseModal()
        navigate("/home")
      }
    })
  }

  return (
    <>
      <Modal open={openModal} onClose={handleCloseModal} className="w-[90%]" disableEscapeKeyDown disablePortal>
        <Box style={styles.modalStyle}>
          <div className="pop-up m-auto bg-white dark:bg-black md:rounded-2xl">
            <button className="relative left-[-80px] top-4 h-10 w-10 rounded-3xl bg-transparent bg-white text-2xl text-black no-underline hover:bg-lightHover dark:bg-black dark:text-white dark:hover:bg-darkHover" onClick={handleCloseModal}>
              x
            </button>

            <img src={darkMode ? darkLogo : lightLogo} alt="GigaChat Logo" className="-mt-4 ml-[45%] w-[40px]" />

            <PreStep handleCloseModal={handleCloseModal} nextShow={nextShow} />

            <FirstStep nickName={nickName} setNickName={setNickName} email={email} setEmail={setEmail} month={month} setMonth={setMonth} day={day} setDay={setDay} year={year} setYear={setYear} nextShow={nextShow} emailExistError={emailExistError} setEmailExistError={setEmailExistError} validEmail={validEmail} mock={mock} />

            <SecondStep nextShow={nextShow} />

            <ThirdStep nickName={nickName} email={email} month={month} day={day} year={year} emailExistError={emailExistError} validEmail={validEmail} mock={mock} nextShow={nextShow} handleOpenBirthdateError={handleOpenBirthdateError} />

            <ForthStep setUser={setUser} setUserToken={setUserToken} nextShow={nextShow} handleOpenBirthdateError={handleOpenBirthdateError} mock={mock} email={email} />

            <FifthStep mock={mock} user={user} setUser={setUser} setUserTag={setUserTag} userToken={userToken} setOriginalUsername={setOriginalUsername} nextShow={nextShow} password={password} setPassword={setPassword} />

            <TagStep mock={mock} userTag={userTag} setUserTag={setUserTag} originalUsername={originalUsername} userToken={userToken} user={user} setUser={setUser} nextShow={nextShow} />

            <UploadProfilePicture userR={user} setUser={setUser} handleCompleteSignup={handleCompleteSignup} handleCloseModal={handleCloseModal} fromSwitch={false} />

            <ErrorPage setDay={setDay} setMonth={setMonth} setYear={setYear} setNickName={setNickName} setEmail={setEmail} openBirthdateErrorModal={openBirthdateErrorModal} handleCloseBirthdateErrorModal={handleCloseBirthdateErrorModal} />
          </div>
        </Box>
      </Modal>
    </>
  )
}
export default SignUp
