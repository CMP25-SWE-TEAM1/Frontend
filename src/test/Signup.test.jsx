import React from "react"

import { screen, render, within, act, fireEvent } from "@testing-library/react"
import SignUp from "../components/Signup/SignUp"
import { BrowserRouter } from "react-router-dom"
import { userEvent } from "@testing-library/user-event"

import { Provider } from "react-redux"
import store from "../store"

import { GoogleOAuthProvider } from "@react-oauth/google"
const clientId = "341526416859-a1u3gf1rl41o6vj5nvl0bs3ac00sljue.apps.googleusercontent.com"

describe("Modal state changes correctly", () => {
  // test("check if modal initially closed", () => {
  //   render(
  //     <BrowserRouter>
  //       <GoogleOAuthProvider clientId={clientId}>
  //         <Provider store={store}>
  //           <Login openModal={false} handleCloseModal={() => {}} setLocation={() => {}} />
  //         </Provider>
  //       </GoogleOAuthProvider>
  //     </BrowserRouter>
  //   )

  //   const closedLoginModal = screen.getByTestId("loginModal")
  //   expect(closedLoginModal).not.toBeInTheDocument()
  // })

  test("Modal can be opened", () => {
    render(
      <BrowserRouter>
        <GoogleOAuthProvider clientId={clientId}>
          <Provider store={store}>
            <SignUp openModal={true} handleCloseModal={() => {}} />
          </Provider>
        </GoogleOAuthProvider>
      </BrowserRouter>
    )

    const openSignupModal = screen.getByTestId("signupModal")
    expect(openSignupModal).toBeInTheDocument()
  })
})

describe("First Page", () => {
  test("Header renders correctly", () => {
    render(
      <BrowserRouter>
        <GoogleOAuthProvider clientId={clientId}>
          <Provider store={store}>
            <SignUp openModal={true} handleCloseModal={() => {}} />
          </Provider>
        </GoogleOAuthProvider>
      </BrowserRouter>
    )

    const header = screen.getByRole("heading", {
      name: /join gigachat today/i,
      hidden: true,
    })
    expect(header).toBeInTheDocument()
  })

  test("Logo renders correctly", () => {
    render(
      <BrowserRouter>
        <GoogleOAuthProvider clientId={clientId}>
          <Provider store={store}>
            <SignUp openModal={true} handleCloseModal={() => {}} />
          </Provider>
        </GoogleOAuthProvider>
      </BrowserRouter>
    )

    const logo = screen.getByRole("img", {
      name: /gigachat logo/i,
      hidden: true,
    })
    expect(logo).toBeInTheDocument()
  })

  test("Google login renders correctly", () => {
    render(
      <BrowserRouter>
        <GoogleOAuthProvider clientId={clientId}>
          <Provider store={store}>
            <SignUp openModal={true} handleCloseModal={() => {}} />
          </Provider>
        </GoogleOAuthProvider>
      </BrowserRouter>
    )

    const googleLogin = screen.getByRole("button", {
      name: /google logo sign up with google/i,
      hidden: true,
    })
    expect(googleLogin).toBeInTheDocument()
  })

  test("Create account renders correctly", async () => {
    render(
      <BrowserRouter>
        <GoogleOAuthProvider clientId={clientId}>
          <Provider store={store}>
            <SignUp openModal={true} handleCloseModal={() => {}} />
          </Provider>
        </GoogleOAuthProvider>
      </BrowserRouter>
    )

    const createAccount = screen.getByRole("button", {
      name: /create account/i,
      hidden: true,
    })
    expect(createAccount).toBeInTheDocument()

    await userEvent.click(createAccount)
  })
})

describe("Second Page", () => {
  test("Name input", async () => {
    render(
      <BrowserRouter>
        <GoogleOAuthProvider clientId={clientId}>
          <Provider store={store}>
            <SignUp openModal={true} handleCloseModal={() => {}} />
          </Provider>
        </GoogleOAuthProvider>
      </BrowserRouter>
    )

    const nameInput = screen.getByTestId("nameInput")
    expect(nameInput).toBeInTheDocument()

    await userEvent.type(nameInput, "Mohamed")
    expect(nameInput.value).toBe("Mohamed")
  })

  test("Email input", async () => {
    render(
      <BrowserRouter>
        <GoogleOAuthProvider clientId={clientId}>
          <Provider store={store}>
            <SignUp openModal={true} handleCloseModal={() => {}} />
          </Provider>
        </GoogleOAuthProvider>
      </BrowserRouter>
    )

    const emailInput = screen.getByTestId("emailInput")
    expect(emailInput).toBeInTheDocument()

    await userEvent.type(emailInput, "Mohamed")
    expect(emailInput.value).toBe("Mohamed")

    // const focusSpy = jest.spyOn(emailInput, "focus")
    // fireEvent.focus(emailInput)
    // expect(focusSpy).toHaveBeenCalled()

    // const blurSpy = jest.spyOn(emailInput, "blur")
    // fireEvent.blur(emailInput)
    //   expect(blurSpy).toHaveBeenCalled()
      
      emailInput.focus()
      emailInput.blur()
    //   jest.spyOn()
  })
})
