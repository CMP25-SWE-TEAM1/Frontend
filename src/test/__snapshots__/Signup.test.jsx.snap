// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage component renders ErrorPage correctly 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="m-auto -mt-10 hidden w-[320px]"
    id="Error Page"
  >
    <div
      class="!h-fit"
    >
      <div
        aria-describedby="modal-modal-description"
        aria-labelledby="modal-modal-title"
        class="relative MuiModal-root css-79ws1d-MuiModal-root"
        role="presentation"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        />
        <div
          data-testid="sentinelStart"
          tabindex="0"
        />
        <div
          class="MuiBox-root css-1atvuul"
          tabindex="-1"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            id="modal-modal-title"
          >
            Can't complete your signup right now.
          </h2>
          <button
            class="btn mt-3 w-[100px] dark:!bg-white"
          >
            Close
          </button>
        </div>
        <div
          data-testid="sentinelEnd"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</div>
`;
